cirq-core/cirq/ops/kraus_channel.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Dict, Iterable, Optional, Tuple
+import numpy as np
+
+from cirq import protocols, value
+from cirq._compat import proper_repr
+from cirq.ops import raw_types
+
+
+class KrausChannel(raw_types.Gate):
+    """A generic channel that can record the index of its selected operator.
+
+    Args:
+        kraus_ops: a list of Kraus operators, formatted as numpy array.
+        key: an optional measurement key string for this channel. Simulations
+            which select a single Kraus operator to apply will store the index
+            of that operator in the measurement result list with this key.
+    """
+
+    def __init__(self, kraus_ops: Iterable[np.ndarray], key: Optional[str] = None):
+        # TODO: validate channel representations (issue #2271)
+        kraus_ops = list(kraus_ops)
+        if not kraus_ops:
+            raise ValueError('KrausChannel must have at least one operation.')
+        num_qubits = np.log2(kraus_ops[0].size) / 2
+        if not num_qubits.is_integer():
+            raise ValueError(
+                f'Input Kraus ops of shape {kraus_ops[0].shape} does not '
+                'represent an operator over qubits.'
+            )
+        self._num_qubits = int(num_qubits)
```

Comment:
Wouldn't it be impossible to get to this point if `num_qubits` wasn't already an integer ? This code seems a little strange/brittle

Reply:
At this point `num_qubits` is an `np.integer`; we want to store a Python `int`.